cobc/pplex.l

```diff
@@ -2613,33 +2649,16 @@ ppecho (const char *text, const cob_u32_t alt_space, const int textlen)
 	fflush (ppout);
 #endif
 
-	/* Check for replacement text before outputting */
-	if (alt_space) {
-		s = yytext;
-	} else {
-		s = text;
-	}
-
 	if (text_queue == NULL && (text[0] == ' ' || text[0] == '\n')) {
-		/* No replacement */
-		fwrite (text, (size_t)textlen, (size_t)1, ppout);
-		/* TODO: instead of \n (empty line: set "needs source-loc" flag and
-		   before first non-empty line output a #line directive, saving
-		   quite some file io [keep 1 empty line]) */
-		if (cb_listing_file) {
-			check_listing (s, 0);
-		}
+		ppecho_direct (text, token);
 		return;
 	}
 	if (!current_replace_list && !base_replace_list) {
 		/* Output queue */
 		for (; text_queue; text_queue = text_queue->next) {
-			fputs (text_queue->text, ppout);
-		}
-		fwrite (text, (size_t)textlen, (size_t)1, ppout);
-		if (cb_listing_file) {
-			check_listing (s, 0);
+			ppecho_direct(text_queue->text, text_queue->token);
```

Comment:
replacing that looks correct; but shockingly we seem to have no single testcase for this - can you please duplicate the test below in syn_copy.at for "preprocess listing" (different expected result), so that we see it is not changed unexpected?